File: PSW-projekat/PZL-Angular/src/app/modules/hospital/model/roomDto.ts
import { Bed } from "./bed";
import { BedDto } from "./bedDto";
import { Blood } from "./blood";
import { Medicine } from "./medicine";
import { RoomType } from "./roomType";

export class RoomDto {
    id: number = 0;
    number: string = '';
    floor: number = 0;
    roomType: RoomType;
    bedDtos: BedDto[];

    public constructor(id: any, number: any, floor: any, roomType: any, bedDtos: any) {
        this.id = id;
        this.number = number;
        this.floor = floor;
        this.roomType = roomType;
        this.bedDtos = bedDtos;

    }

    
}

File: PSW-projekat/hospital-be-main/src/HospitalLibrary/Core/DTOs/RoomDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using HospitalLibrary.Core.Model.Enums;

namespace HospitalLibrary.Core.DTOs
{
    public class RoomDto
    {
        public int Id { get; set; }
        public string Number { get; set; }
        [Range(1, 10)]

        public int Floor { get; set; }
        public RoomType RoomType { get; set; }
        public virtual ICollection<BedDto> BedDtos { get; set; }

        public RoomDto() { }
        public RoomDto(int id, string number, int floor, RoomType roomType, ICollection<BedDto> bedDtos)
        {
            Id = id;
            Number = number;
            Floor = floor;
            RoomType = roomType;
            BedDtos = bedDtos;
        }
    }
}
